Neos.Fusion/Classes/Core/FusionSourceCode.php

```diff
@@ -0,0 +1,61 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Fusion\Core;
+
+/*
+ * This file is part of the Neos.Fusion package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+
+#[Flow\Proxy(false)]
+class FusionSourceCode
+{
+    private function __construct(
+        private string $sourceCode,
+        private ?string $contextPathAndFilename
+    ) {
+        if (trim($sourceCode) === '') {
+            throw FusionSourceCodeIsInvalid::becauseTheSourceCodeIsEmpty($this->contextPathAndFilename);
```

Comment:
Please, no static "factories" for exceptions!

Reply:
fine with me ;)

Reply:
Haven't seen such a way to instantiate an exception before. My gut feeling says: unnecessarily fancy / complicated. But maybe it's also just unfamiliar.

Reply:
okay ;)
https://github.com/neos/neos-development-collection/pull/3839#discussion_r991122895

lets move the discussion to https://discuss.neos.io/t/rfc-updated-default-code-style/5836?u=marc